Compilation failure when generating builders for a class containing both `http` and `https` fields

It seems weird to me that I am the first one hitting this, so I _could definitely_ be missing something, but anyway... let's have the following reproducer:

1. pom.xml:
```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <groupId>org.jboss.experiments</groupId>
    <artifactId>sundrio-test-http-s</artifactId>
    <version>1.0-SNAPSHOT</version>

    <properties>
        <maven.compiler.source>17</maven.compiler.source>
        <maven.compiler.target>17</maven.compiler.target>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    </properties>

    <dependencies>
        <dependency>
            <groupId>io.sundr</groupId>
            <artifactId>builder-annotations</artifactId>
            <version>0.220.1</version>
        </dependency>
    </dependencies>

</project>
```

2. `MyBuildableClass` class:
```
@io.sundr.builder.annotations.Buildable(editableEnabled = false)
public class MyBuildableClass {

    private java.util.List<org.jboss.experiments.Http> http;

    public java.util.List<org.jboss.experiments.Http> getHttp() {
        return http;
    }

    public void setHttp(java.util.List<org.jboss.experiments.Http> http) {
        this.http = http;
    }

    private java.util.List<org.jboss.experiments.Https> https;

    public java.util.List<org.jboss.experiments.Https> getHttps() {
        return https;
    }

    public void setHttps(java.util.List<org.jboss.experiments.Https> https) {
        this.https = https;
    }
}
```

3. `Http` class
```
public class Http {}
```

4. `Https` class
```
public class Https {}
```

`mvn clean package` fails with Sundrio generating duplicate methods:

```
...
[ERROR] /home/fburzigo/projects/git/fabiobrz/sundrio-test-http-s/target/generated-sources/annotations/org/jboss/experiments/MyBuildableClassFluent.java:[217,16] method getHttp(int) is already defined in class org.jboss.experiments.MyBuildableClassFluent
[ERROR] /home/fburzigo/projects/git/fabiobrz/sundrio-test-http-s/target/generated-sources/annotations/org/jboss/experiments/MyBuildableClassFluent.java:[221,16] method getFirstHttp() is already defined in class org.jboss.experiments.MyBuildableClassFluent
[ERROR] /home/fburzigo/projects/git/fabiobrz/sundrio-test-http-s/target/generated-sources/annotations/org/jboss/experiments/MyBuildableClassFluent.java:[225,16] method getLastHttp() is already defined in class org.jboss.experiments.MyBuildableClassFluent
[ERROR] /home/fburzigo/projects/git/fabiobrz/sundrio-test-http-s/target/generated-sources/annotations/org/jboss/experiments/MyBuildableClassFluent.java:[229,16] name clash: getMatchingHttp(java.util.function.Predicate<org.jboss.experiments.Https>) and getMatchingHttp(java.util.function.Predicate<org.jboss.experiments.Http>) have the same erasure
[ERROR] /home/fburzigo/projects/git/fabiobrz/sundrio-test-http-s/target/generated-sources/annotations/org/jboss/experiments/MyBuildableClassFluent.java:[238,18] name clash: hasMatchingHttp(java.util.function.Predicate<org.jboss.experiments.Https>) and hasMatchingHttp(java.util.function.Predicate<org.jboss.experiments.Http>) have the same erasure
...
```

Where does this come from? I tried to generate a Java object model from Keycloak CRDs, which spec contains both `http` and `https` properties:
- https://github.com/keycloak/keycloak-k8s-resources/blob/26.2.5/kubernetes/keycloaks.k8s.keycloak.org-v1.yml#L186
- https://github.com/keycloak/keycloak-k8s-resources/blob/26.2.5/kubernetes/keycloaks.k8s.keycloak.org-v1.yml#L311

--- a/core/src/main/java/io/sundr/functions/Pluralize.java
+++ b/core/src/main/java/io/sundr/functions/Pluralize.java
@@ -61,7 +61,8 @@ public enum Pluralize implements Function<String, String> {
       new StringReplace("((a)naly|(b)a|(d)iagno|(p)arenthe|(p)rogno|(s)ynop|(t)he)ses$", "$1$2sis"),
       new StringReplace("([ti])um$", "$1a"),
       new StringReplace("(n)ews$", "$1ews"),
-      new StringReplace("(s|si|u)s$", "$1s"));
+      new StringReplace("(s|si|u)s$", "$1s"),
+      new StringReplace("(http|https)$", "$1"));
 
   public String apply(String word) {
     if (word == null) {

--- a/core/src/main/java/io/sundr/functions/Singularize.java
+++ b/core/src/main/java/io/sundr/functions/Singularize.java
@@ -70,6 +70,7 @@ public enum Singularize implements Function<String, String> {
       new StringReplace("([ti])a$", "$1um"),
       new StringReplace("(n)ews$", "$1ews"),
       new StringReplace("(s|si|u)s$", "$1s"),
+      new StringReplace("(http|https)$", "$1"),
       new StringReplace("s$", ""));
 
   public String apply(String word) {

--- a/core/src/test/java/io/sundr/functions/PluralizeTest.java
+++ b/core/src/test/java/io/sundr/functions/PluralizeTest.java
@@ -30,6 +30,10 @@ public class PluralizeTest {
     assertEquals("vertices", Pluralize.FUNCTION.apply("vertex"));
     assertEquals("endpoints", Pluralize.FUNCTION.apply("endpoints"));
     assertEquals("ingresses", Pluralize.FUNCTION.apply("ingress"));
+    assertEquals("http", Pluralize.FUNCTION.apply("http"));
+    assertEquals("Http", Pluralize.FUNCTION.apply("Http"));
+    assertEquals("https", Pluralize.FUNCTION.apply("https"));
+    assertEquals("Https", Pluralize.FUNCTION.apply("Https"));
   }
 
 }

--- a/core/src/test/java/io/sundr/functions/SingularizeTest.java
+++ b/core/src/test/java/io/sundr/functions/SingularizeTest.java
@@ -33,5 +33,11 @@ public class SingularizeTest {
 
     assertEquals("child", Singularize.FUNCTION.apply("children"));
     assertEquals("Child", Singularize.FUNCTION.apply("Children"));
+
+    assertEquals("https", Singularize.FUNCTION.apply("https"));
+    assertEquals("Https", Singularize.FUNCTION.apply("Https"));
+
+    assertEquals("http", Singularize.FUNCTION.apply("http"));
+    assertEquals("Http", Singularize.FUNCTION.apply("Http"));
   }
 }

